randobot/handler.py

```diff
@@ -955,17 +964,41 @@ async def handle_random_seed(self, encrypt, dev, reply_to):
             encrypt=encrypt,
             dev=True,
             reply_to=reply_to,
-            settings=drafted_settings.get('data')
+            settings=self.patch_settings(dev=True)
         )
         if draft.get('race_type') == 'qualifier':
             await self.send_message(
-                f'{reply_to}, use !settings 5 minutes before race start to reveal the selected settings.'
+                f'Race Monitors, use !settings 5 minutes before race start to reveal the selected settings.'
             )
             await self.send_message(
                 'Once revealed, the room will be locked from joining.'
             )
             return
         await self.ex_settings('', '')
 
+    def patch_settings(self, dev):
+        preset = self.zsr.presets.get('s7').get('settings') if not dev else self.zsr.presets_dev.get('s7').get('settings')
```

Comment:
I think this still falls foul of the mutability issue even though it's not in the same class. Safest way to ensure you have a "clean" dict is to use the splat operator, i.e. 

```
settings = self.zsr.presets.get('s7').get('settings') if not dev else self.zsr.presets_dev.get('s7').get('settings')
preset = {**settings}
# ...
preset.update(data)
return preset
```